openedx/openedx-platform
# Add endpoint to get SAML providers for a user.

Adds an endpoint to allow mobile apps to get SSO providers for a given user.  Includes throttling to prevent the endpoint from being used to crawl for logins.  Due to policy concerns around potential leakage of user info, this is hidden behind a feature flag.

Upstreamed from edx-solutions/edx-platform#1047 and edx-solutions/edx-platform#1080

**JIRA tickets**: Implements OC-4285.

**Dependencies**: None

**Screenshots**:

**Sandbox URL**: https://pr18237.sandbox.opencraft.hosting/

**Merge deadline**: None

**Testing instructions**:

Set ALLOW_UNPRIVILEGED_SSO_PROVIDER_QUERY to true in lms.env.json.

Verify that the SSO user endpoint can be accessed by non-logged in users. Example: https://pr18237.sandbox.opencraft.hosting/api/third_party_auth/v0/users/staff

Verify that provider information can be retrieved by providing an email address. Example: https://pr18237.sandbox.opencraft.hosting/api/third_party_auth/v0/users/staff@example.com

**Author notes and concerns**:

**Reviewers**

**Settings**
Set ALLOW_UNPRIVILEGED_SSO_PROVIDER_QUERY to true in lms.env.json.
